File: Core/Models/Word.cs
using Core.Helpers;
using System.ComponentModel.DataAnnotations;

namespace Core.Models;

public class Word : BaseEntity
{
    [Required]
    public string Text { get; set; } = null!;
    public Language Language { get; set; } = Language.EN;
    public int? TranslateId { get; set; }
    public Translate Translate { get; set; }
}

File: Client/src/app/models/word.ts
import { Translate } from "./translate";

export class Word {
    text: string;
    id: number;
    translate: Translate;
    learnt: boolean = false;
    /**
     *
     */
    constructor(text: string, id: number, translate: Translate) {
        this.text = text;
        this.id = id;
        this.translate = translate;
    }
}
